qadence/blocks/primitive.py

```diff
@@ -200,11 +207,10 @@ def _from_dict(cls, d: dict) -> ParametricBlock:
         target = d["qubit_support"][0]
         return cls(target, params)  # type: ignore[call-arg]
 
-    def dagger(self) -> ParametricBlock:  # type: ignore[override]
+    def dagger(self) -> ParametricBlock:
         exprs = self.parameters.expressions()
-        args = tuple(-extract_original_param_entry(param) for param in exprs)
-        args = args if -1 in self.qubit_support else (*self.qubit_support, *args)
-        return self.__class__(*args)  # type: ignore[arg-type]
+        params = tuple(-extract_original_param_entry(param) for param in exprs)
+        return type(self)(*self.qubit_support, *params)  # type: ignore[arg-type]
```

Comment:
I know this is just implementing the same logic that was there before, but this assumes that all `ParametricBlocks` are implemented as `exp(i * param * some_block)`. I guess that is currently true right? Just something to keep in mind, I don't think I have a better solution.

Reply:
yeah, the idea is that the "special" cases just override this method